x-pack/plugins/ml/public/data_frame/components/define_pivot/define_pivot_summary.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { SFC, useContext } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiComboBoxOptionProps,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiText,
+} from '@elastic/eui';
+
+import { AggListSummary } from '../../components/aggregation_list';
+import { GroupByList } from '../../components/group_by_list/list';
+import { PivotPreview } from './pivot_preview';
+
+import { Dictionary } from '../../../../common/types/common';
+import {
+  DropDownLabel,
+  DropDownOption,
+  IndexPatternContext,
+  Label,
+  OptionsDataElement,
+  pivotSupportedAggs,
+  SimpleQuery,
+} from '../../common';
+
+const defaultSearch = '*';
+const emptySearch = '';
+
+interface Props {
+  search: string;
+  groupBy: Label[];
+  aggList: Label[];
+}
+
+export const DefinePivotSummary: SFC<Props> = ({ search, groupBy, aggList }) => {
+  const indexPattern = useContext(IndexPatternContext);
+
+  if (indexPattern === null) {
+    return null;
+  }
+
+  const fields = indexPattern.fields
+    .filter(field => field.aggregatable === true)
+    .map(field => ({ name: field.name, type: field.type }));
+
+  // The available fields for group by
+  const groupByOptions: EuiComboBoxOptionProps[] = [];
+  fields.forEach(field => {
+    const o: DropDownLabel = { label: field.name };
+    groupByOptions.push(o);
+  });
+
+  // The available aggregations
+  const aggOptions: EuiComboBoxOptionProps[] = [];
+  const aggOptionsData: Dictionary<OptionsDataElement> = {};
+
+  fields.forEach(field => {
+    const o: DropDownOption = { label: field.name, options: [] };
+    pivotSupportedAggs.forEach(agg => {
+      if (
+        (agg === 'cardinality' && (field.type === 'string' || field.type === 'ip')) ||
```

Comment:
As before, aggregation and field `type` enums would be useful here

Reply:
Done in 6b4b1f4.